algorithms/src/Kokkos_Sort.hpp

```diff
@@ -704,6 +726,10 @@ template <class ExecutionSpace, class ViewType>
 std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
     const ExecutionSpace& exec, ViewType view, size_t const begin,
     size_t const end) {
```

Comment:
(Not expecting you to resolve here)
Anyone knows why we provide that overload?
Why is it not sufficient to sort a subview?  Also why is it not deferring to the overload that does not take the begin and end range arguments?
@crtrott

Reply:
because this is calling the bin sort which has a method taking a "range" defined by indices.

Reply:
and it works for dynamic views too. 
my plan is to revisit all this in a separate PR to make it more clear why we hae some things and what happens.

Reply:
My point is that if sorting a subview has the same effect, what is the point of this "non-standard" overload?

Reply:
because you need to have the begin/end info to access the correct permutation elements that are stored inside binsort.

Reply:
This makes sense in the context of BinSort but not for the Kokkos::sort algorithm.

Reply:
i totally agree with you. In fact, this seems like this part of the sort api was designed with the assumption that underneath was going to call bin sort.

Reply:
Yeah lets revisit, but not here.

Reply:
yes not here, i will work on a new PR (I already started)